tests/sweep_framework/sweep_utils/conv2d_common.py

```diff
@@ -178,51 +178,111 @@ def run_conv2d_short_sweep(
     input_specs,
     device,
 ) -> list:
-    [
-        batch_size,
-        output_channels,
-        input_channels,
-        input_height,
-        input_width,
-        kernel_height,
-        kernel_width,
-        stride_h,
-        stride_w,
-        pad_h,
-        pad_w,
-        groups,
-        has_bias,
-        dilation,
-    ] = input_specs
+    # for tt-forge suite, extra arguments are tensor configs
+    is_forge_suite = False
+    if len(input_specs) > 15:
+        [
+            batch_size,
+            output_channels,
+            input_channels,
+            input_height,
+            input_width,
+            kernel_height,
+            kernel_width,
+            stride_h,
+            stride_w,
+            pad_h,
+            pad_w,
+            groups,
+            dilation_h,
+            dilation_w,
+            has_bias,
+            [input_layout, input_buffer_type, input_dtype],
+            [weights_layout, weights_buffer_type, weights_dtype],
+            [output_layout, output_buffer_type, output_dtype],
+        ] = input_specs
+        is_forge_suite = True
+    else:
+        [
+            batch_size,
+            output_channels,
+            input_channels,
+            input_height,
+            input_width,
+            kernel_height,
+            kernel_width,
+            stride_h,
+            stride_w,
+            pad_h,
+            pad_w,
+            groups,
+            dilation_h,
+            dilation_w,
+            has_bias,
+        ] = input_specs
     print(input_specs)
 
+    if is_forge_suite:
+        torch_input_dtype = torch.bfloat16 if input_dtype == ttnn.DataType(ttnn.bfloat16) else torch.float32
+        torch_weight_dtype = torch.bfloat16 if weights_dtype == ttnn.DataType(ttnn.bfloat16) else torch.float32
+
     conv_input_shape = [batch_size, input_channels, input_height, input_width]
     conv_weight_shape = [output_channels, input_channels // groups, kernel_height, kernel_width]
     conv_bias_shape = [1, 1, 1, output_channels]
-    torch_input_tensor_nchw = torch.randn(conv_input_shape, dtype=torch.bfloat16).float()
+    torch_input_tensor_nchw = torch.randn(
+        conv_input_shape, dtype=torch_input_dtype if is_forge_suite else torch.bfloat16
```

Comment:
can we introduce dtype variables for input, weights, and bias and initalize them differently if it's forge/pytorch sweep and use them when we create tensors

Reply:
It is already there, line no. 200-202, we are deriving them from forge suite. For short sweep all are torch.bfloat16.
For bias, we use same datatype as weights in our test cases also. so is here.

Reply:
` dtype=torch_input_dtype if is_forge_suite else torch.bfloat16`
we can just put dtype here which is figured out before

Reply:
This check is for short sweep. For short sweep, we are using only bfloat16.